### PR TITLE
Try fix flaky test test_parallel_distributed_insert_select_with_schema_inference

### PR DESCRIPTION
**Pull request by Avogar**
### Changelog category (leave one):

- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: tests/integration/test_s3_cluster/test.py [hunk 1]
```diff
@@ -32,8 +32,9 @@ def create_buckets_s3(cluster):
             # a String, b UInt64
             data = []
 
-            for number in range(100):
-                data.append([str(number) * 10, number])
+            # Make all files a bit different
```
Comment (antonio2368) [inline tests/integration/test_s3_cluster/test.py:35]: why?

Comment (Avogar) [reply tests/integration/test_s3_cluster/test.py:35]: My hypothesis: 
We have 100 identical files with the same data. During parallel insert select from s3Cluster each replica inserts only 50/100 files (for example first inserts from 1-50 files, second from 51-100). And when insert is finished, both replicas contains identical 5000 rows and for some reason we think that that's all and we don't need to replicate this data between replicas to complete it to 10000 rows (maybe we check checksums of data parts and see that both replicas have the same set of parts). I am not very familiar with replication process, but this looks like the reason (after my change it's not flaky locally)

Comment (Avogar) [reply tests/integration/test_s3_cluster/test.py:35]: Making 99 files instead of 100 files also helps locally

**Review by antonio2368 — APPROVED** [review]

File: tests/integration/test_s3_cluster/test.py (hunk 1 above)
Comment (antonio2368) [reply tests/integration/test_s3_cluster/test.py:35]: Good thing we have the flaky check, the failure occurs relatively often in CI so it should be a good indication that it's okay now.

Comment (Avogar) [reply tests/integration/test_s3_cluster/test.py:35]: Just got a confirmation that it's block deduplication process.

**Comment by antonio2368** [comment]
Going to merge because flaky check is okay and test failures are unrelated.

**Pull request merged**